app/services/importer/get_workday_fields.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+module Importer
+  class GetWorkdayFields < ApplicationTask
+    # Some Workday companies require a login cookie to fetch fields
+    # If login is required, the API response will include an authentication widget
+    # #user_authenticated? checks this and logs in using ENV credentials if required
+    # WorkdayCookieFetcher retrieves the cookie, which is attached to the request header
+    # Then we make the request again, attaching the cookie
```

Comment:
Can you also explain the workings of this file - mostly explains the cookie fetcher at the moment (which is also super helpful

Reply:
Done!